### DISCUSSION TITLE
Custom keybinds via config?

### DISCUSSION THREAD

**Discussion by jeyj0** [PatrickF1/fzf.fish#89] [Archive] [answered]
Hello, really nice plugin you got here!

I keep all my configs synced via git, which is why I don't like to configure something using fish's universal vars. Do you think a way to configure using a file might be possible? I'd actually prefer it if this plugin wouldn't map any bindings and just included an example config to add yourself.

Another option for me would be to change the Ctrl+l binding to Ctrl+g to not interfere with the "clear" binding which is ctrl+l quite often. But that doesn't seem as long-lasting as adding a configuration option.

**Comment by PatrickF1** [comment]
Hi Jannis! Just to be clear, what you want is to disable the default keybindings in your `config.fish` right? If so, thanks for bringing up this use case, which I hadn’t thought of! Let me investigate this more and get back to you. In the meantime, I believe some people check in their fish universal variables (the file `~/.config/fish/fish_variables`). Do you think that might work for you?

For what it's worth by the way, the other fzf plugin has the same problem: https://github.com/jethrokuan/fzf#usage. Notice the instructions say to disable default keybindings using universal variables.

**Comment by PatrickF1** [comment]
Hi again @jeyj0 , I have done more research and it seems it is currently impossible for conf.d scripts to execute AFTER config.fish. I was afraid of this. I am getting this from https://fishshell.com/docs/current/#initialization-files, which lists `config.fish` as the last phase of initialization. I opened up an issue with fish-shell to double check my understanding and ask for alternatives. fish-shell/fish-shell#7256

**Comment by PatrickF1** [comment]
Ok after more research there is a hack that is possible via erasing the key bindings with an event trigger on `fzf_fish_custom_keybindings`. However, it is hacky, difficult to maintain, and can have problems so I'm not going to do it, sorry :(

Instead, here are two possibilities for those who want to check in setting `fzf_fish_custom_keybindings` in their config.fish:
- just execute `set -U fzf_fish_custom_keybindings`; it's idempotent anyway
- check in a file `~/.config/fish/conf.d/0.fzf` which will be executed before `fzf.fish` setting `fzf_fish_custom_keybindings`

**Comment by jeyj0** [comment]
Hi Patrick, thanks for all your efforts! I didn't know it's that easy to simply check-in universal variables in fish. That actually solves the issue for me already. Thanks again for all this effort though!

**Comment by PatrickF1** [comment, marked as answer]
@jeyj0 With https://github.com/PatrickF1/fzf.fish/releases/tag/v7.0, custom key bindings have now been overhauled so you can check in a function call to your config.fish without using universal variables (which don't belong in your config.fish).
